pkg/resources/kafka/allBrokerService.go

```diff
@@ -64,67 +53,3 @@ func (r *Reconciler) allBrokerService() runtime.Object {
 		},
 	}
 }
-
-// deleteNonHeadlessServices deletes the all-broker service that was created for the current KafkaCluster
-// if there is any and also the service of each broker
-func (r *Reconciler) deleteNonHeadlessServices() error {
```

Comment:
I would not move this function from the allBrokerService because basicly deleteNonHeadlessServices = deleteAllBrokerService and it is more connected to the func (r *Reconciler) allBrokerService() function.
I would rename the deleteNonHeadlessServices to deleteAllBrokerService.

We can se the same for the headlessService - deleteHeadlessService() in the headlessService.go

Reply:
Well, if you look into the details of this function, it is not just to delete the all broker service, it also deletes the services like the `NodePort` services, and LoadBalancer (the implementation for this will be needed in another PR)

Reply:
You have right regarding this function deletes not only the all broker service but deletes those services which belongs to all broker services (https://github.com/banzaicloud/koperator/blob/71000a41df4fe3c1565217df9ff75ae9b9adbec6/pkg/resources/kafka/kafka.go#L385)
Then I would say deleteAllBrokerServices not deleteAllBrokerService.
I understand your logic.  I think both can be good.

Reply:
Actually I think the `allBrokerService.go` is meant for the `"%s-all-broker"` service only, not the "services for all the brokers". That's why I don't think it makes sense to put anything that is not related to the `"%s-all-broker"` service to that file

> I know this is confusing 😕

Reply:
It is a little bit weird "non headless serices for all brokers" :)

Reply:
agreed, I can update it to "failed to delete non-headless services for all of the brokers" to make it less confusing

Reply:
log message updated

Reply:
Thank you! :)